yodeploy/test_integration/samples/uwsgi-app/deploy/hooks.py

```diff
@@ -0,0 +1,20 @@
+import getpass
+import os
+
+from mock import Mock
+
+from yodeploy.hooks.uwsgi import UwsgiHostedApp
+from yodeploy.test_integration.helpers import tests_dir
+
+
+class Hooks(UwsgiHostedApp):
+    # these overrides are only needed for testing
+    logs_path = os.path.join(tests_dir, 'filesys/var/log/uwsgi')
+    config_path = os.path.join(tests_dir, 'filesys/etc/uwsgi')
+    user = getpass.getuser()
+    group = None
+
+    _reload_uwsgi = Mock()
```

Comment:
Thinking of tests as the first consumers of the code, if short circuiting uwsgi is useful then the method should probably be public, not private.

Reply:
Slightly disagree here.  IMO tests shouldn't prevent us from making method/function protected/private.  Otherwise, we won't have protected/private functions/methos at all.

Reply:
I heard one opinion (which I think is slowly starting to become my own) that if your language doesn't support public/private members, don't try to pretend like that feature exists. Because even if your member is private, somewhere/someone/something will consume it (as we see in this test).

So embrace it. Make it public and don't worry about 🙈🙉🎉

Reply:
Yeah, from technical point of view - we don't have private methods :)  But it's kinda convenient to say all other developers about your intention.  IMO this helps a lot to understand the code.  Thus let me rephrase your proposal a bit:  don't worry about mocking/testing so-called private/protected methods/functions in python :)

Reply:
Also, technically variable names don't matter at all. Which doesn't mean we shouldn't spent hours thinking of them.